vgaidarji/react-graphql-playground
# Move to monorepo setup

### What has been done
1. Moved from a single application project structure to mono-repo-like project structure using NPM workspaces. 
   	```
	packages \
	    api - future package with API for population snapshot data to be consumed by frontend app
	    frontend - react app
	    e2e - cypress tests
	```
2. ESLint and Prettier are now applied on the root level and evaluate all files in all packages.
3. Code coverage is only configured for `frontend` package. `api` will get tests in future PRs and will need to publish + distinguish coverages from packages using codecov tagging.

Use specific script commands from each package 
`cd packages/frontend && npm run test`
or run commands (if defined) from project root on all workspaces as follows
`npm run test --workspaces --if-present`

### How to test
1. Run various CLI commands locally to test the application functioning and all scripts working. 
2. Confirm CI is passing and all steps are working as before (building, testing, code coverage, deploy to firebase)
